### PR TITLE
Adding pause feature to Hedera Token Service

### PR DESCRIPTION
**Pull request by chung-chao**
**Description**:

**Related issue(s)**:

Fixes #

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by sergmetelin — CHANGES_REQUESTED** [review]
@chung-chao looks good to me, but you need to add the DCO (Developer Certificate of Ownership) signoff on the commits in the PR to pass the checks. 

https://github.com/hashgraph/hedera-improvement-proposal/pull/132/checks?check_run_id=3468345061

**Comment by sergmetelin** [comment]
could you also rename the file to hip-24.md?

File: HIP/hip-0000-adding-pause–feature-to-HTS.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
+- hip: [TBD]
```
Comment (sergmetelin) [inline HIP/hip-0000-adding-pause–feature-to-HTS.md:1]: Please update to HIP-24

File: HIP/hip-0000-adding-pause–feature-to-HTS.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+- title: Pause feature on Hedera Token Service
+- author: chung@xfers.com, junhao@xfers.com, zhikai@xfers.com
+- type: Standards Track
+- category: Application
+- status: Draft
+- created: 2021-08-09
+- discussions-to:
```
Comment (sergmetelin) [inline HIP/hip-0000-adding-pause–feature-to-HTS.md:8]: Please add a discussion link to https://github.com/hashgraph/hedera-improvement-proposal/discussions/126

File: HIP/hip-0000-adding-pause–feature-to-HTS.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+1. Compliance with regulatory frameworks across jurisdictions in the case of regulated token issuance (e.g. stablecoins)
+2. Compliance with law enforcement requests
+3. Parity with functionalities offered by other blockchain protocols in the case of multichain asset issuance.
+
+We propose to establish a standard for HTS where a pause feature can be included for tokens deployed via HTS. This will be opt-in and allows token issuers the flexibility of including or excluding the pause functionality based on the token issuers' requirements. 
+
+## **Specification**
+
+To be Completed.
```
Comment (steven-sheehy) [inline HIP/hip-0000-adding-pause–feature-to-HTS.md:27]: Should be completed before merged.

File: HIP/hip-0000-adding-pause–feature-to-HTS.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        uint64 initialSupply = 4; // Specifies the initial supply of tokens to be put in circulation. The initial supply is sent to the Treasury Account. The supply is in the lowest denomination possible. In the case for NON_FUNGIBLE_UNIQUE Type the value must be 0
+        AccountID treasury = 5; // The account which will act as a treasury for the token. This account will receive the specified initial supply or the newly minted NFTs in the case for NON_FUNGIBLE_UNIQUE Type
+        Key adminKey = 6; // The key which can perform update/delete operations on the token. If empty, the token can be perceived as immutable (not being able to be updated/deleted)
+        ...
+        ...
+        ...
+        Key fee_schedule_key = 20; // The key which can change the token's custom fee schedule; must sign a TokenFeeScheduleUpdate transaction
+        repeated CustomFee custom_fees = 21; // The custom fees to be assessed during a CryptoTransfer that transfers units of this token
+        Key pauseKey = 22; // [New] The key which can pause the token. If empty, pause is not possible
```
Comment (steven-sheehy) [inline HIP/hip-0000-adding-pause–feature-to-HTS.md:44]: This sets up permissions for the pause action, but how does the actual pause and unpause take place?

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: HIP/hip-0000-adding-pause–feature-to-HTS.md (hunk 3 above)
Comment (chung-chao) [reply HIP/hip-0000-adding-pause–feature-to-HTS.md]: Noted. Would finalize that before the end of next Monday, SGT

File: HIP/hip-0000-adding-pause–feature-to-HTS.md (hunk 4 above)
Comment (chung-chao) [reply HIP/hip-0000-adding-pause–feature-to-HTS.md]: Noted. Would finalize that before the end of next Monday, SGT

**Comment by chung-chao** [comment]
Also, noted with HIP number and commits for signoff. Thanks for your information.

File: HIP/hip-0000-adding-pause–feature-to-HTS.md (hunk 4 above)
Comment (chung-chao) [reply HIP/hip-0000-adding-pause–feature-to-HTS.md]: @steven-sheehy Thanks for your suggesting. We've updated with the specification. Hope that the question is covered in the commit.

**Review by sergmetelin — APPROVED** [review]
LGTM

**Review by steven-sheehy — APPROVED** [review]

**Pull request merged**